packages/nimble-components/src/wafer-map/modules/data-manager.ts

```diff
@@ -54,27 +54,43 @@ export class DataManager {
         return this.prerendering.diesRenderInfo;
     }
 
-    private readonly computations: Computations;
-    private readonly prerendering: Prerendering;
-    private readonly dataMap: Map<string, WaferMapDie>;
-
-    public constructor(wafermap: WaferMap) {
-        this.computations = new Computations(wafermap);
-
-        this.prerendering = new Prerendering(
-            wafermap,
-            this.horizontalScale,
-            this.verticalScale,
-            this.dieDimensions,
-            this.margin
-        );
+    public get data(): Map<string, WaferMapDie> {
+        return this.dataMap;
+    }
 
-        this.dataMap = new Map(
-            wafermap.dies.map(die => [`${die.x}_${die.y}`, die])
-        );
+    private readonly computations = new Computations(this.wafermap);
+    private readonly prerendering = new Prerendering(this.wafermap, this);
+    private dataMap!: Map<string, WaferMapDie>;
+
+    public constructor(private readonly wafermap: WaferMap) {}
```

Comment:
Similar to the discussion in the event-coordinator, because creating Computations and Prerendering rely on the constructor running first we should just make those members in the constructor.